packages/framework/src/hooks/useStyleNames.ts

```diff
@@ -1,28 +1,56 @@
 import { useAtomValue } from "jotai";
 import type { CSSProperties } from "react";
-import { type Expression, resolveStyleNames } from "../shared";
+import { get, intersection, isEmpty, isString, keys, merge } from "lodash-es";
+import { type EnsembleThemeModel, type Expression } from "../shared";
 import { themeAtom } from "../state";
-import { isString } from "lodash-es";
+import { useEvaluate } from "./useEvaluate";
+
+const resolveStyleNames = (
+  styleNames: string[],
+  theme: EnsembleThemeModel,
+): CSSProperties => {
+  const appStyleNames = keys(theme.Styles);
+  const overlappingNames = intersection(styleNames, appStyleNames);
+  if (isEmpty(overlappingNames)) {
+    return {};
+  }
+
+  const appliedStyles = {};
+  overlappingNames.forEach((name) => {
+    merge(appliedStyles, get(theme.Styles, name));
+  });
+
+  return appliedStyles;
+};
 
 export const useStyleNames = (
   namedStyles?: Expression<string | string[]>,
   classStyles?: Expression<string | string[]>,
 ): CSSProperties | undefined => {
   const themeContext = useAtomValue(themeAtom);
 
-  let dotClassStyles = "";
-  if (isString(classStyles) && classStyles.length > 0) {
-    dotClassStyles = classStyles
-      .trim()
-      .split(" ")
-      .map((className) => `.${className}`)
-      .join(" ");
-  }
-  const styleNames = `${
-    isString(namedStyles) ? namedStyles : ""
-  } ${dotClassStyles}`;
+  // evaluate styles
+  const nameStylesArray: string[] | undefined = isString(namedStyles)
+    ? namedStyles.split(" ")
+    : namedStyles;
+  const nameStyles = useEvaluate({ nameStylesArray });
+
+  // evaluate classnames
+  const classStylesArray: string[] | undefined = isString(classStyles)
+    ? classStyles.split(" ")
+    : classStyles;
+  const styleClasses = useEvaluate({ classStylesArray });
+
+  const styleNames = [
+    ...(nameStyles.nameStylesArray
+      ? nameStyles.nameStylesArray.map((nameStyle) => `${nameStyle}`)
+      : []),
```

Comment:
You don't need both the spread operator and map. Both will create new arrays.

Reply:
this is for merge the names styles and class styles in single array